scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py

```diff
@@ -140,4 +148,38 @@ def update_view_from_model(self):
         for group in self._model.groups:
             self.add_group_to_view(group)
 
+        self.from_file_checkbox_changed()
+        self.first_good_data_checkbox_changed()
         self._view.enable_updates()
+
+    def first_good_data_checkbox_changed(self):
+        if self._view.group_range_use_first_good_data.isChecked():
+            if self._model._data.current_runs:
+                self._view.group_range_min.setText(str(self._model._data.loaded_data(self._model._data.current_runs[-1])["FirstGoodData"]))
+
+            self._view.group_range_min.setEnabled(False)
+            if 'GroupRangeMin' in self._model._data.gui_variables:
+                self._model._data.gui_variables.pop('GroupRangeMin')
+        else:
+            self._view.group_range_min.setEnabled(True)
+            self._model._data.gui_variables['GroupRangeMin'] = float(self._view.group_range_min.text())
+
+    def from_file_checkbox_changed(self):
+        if self._view.group_range_use_last_data.isChecked():
+            if self._model._data.current_runs:
+                self._view.group_range_max.setText(str(round(max(self._model._data.loaded_data(self._model._data.
+                                                                                               current_runs[-1])['OutputWorkspace'][0].
+                                                                 workspace.dataX(0)), 3)))
```

Comment:
whats the significance of the 3?

Reply:
This is the number of decimal places to round to before displaying

Reply:
could we make it a variable with a descriptive name?